src/sync/sync-utils.ts

```diff
@@ -49,6 +54,57 @@ export async function findOrStartSync(
 	}, 0, logger);
 }
 
+type subscriptionUpdateTasks = {
+	totalNumberOfRepos?: number | null;
+	repositoryCursor?: string | null;
+	repositoryStatus?: string | null;
+}
+
+const resetTargetedTasks = async (subscription: Subscription, syncType?: SyncType, targetTasks?: TaskType[]): Promise<void> => {
+	if (!syncType || !targetTasks?.length) {
+		return;
+	}
+
+	const subscriptionTasks = targetTasks.find(task => task === "repository");
+	const repoSyncTasks = targetTasks.filter(task => task !== "repository");
+	const updateRepoSyncTasks = {};
+	const updateSubscriptionTasks: subscriptionUpdateTasks = {};
+
+	// Reset RepoSync states - target tasks: ("pull" | "commit" | "branch" | "build" | "deployment")
+	// Full sync resets cursor and status
+	// Partial sync only resets status (continues from existing cursor)
+	repoSyncTasks.forEach(task => {
+		if (syncType === "full") {
+			updateRepoSyncTasks[`${task}Cursor`] = null;
+		}
+		updateRepoSyncTasks[`${task}Status`] = null;
+	});
+
+	await RepoSyncState.update({
+		repoUpdatedAt: null,
+		...updateRepoSyncTasks
+	}, {
+		where: {
+			subscriptionId: subscription.id
+		}
+	});
+
+	// Reset Subscription Repo state -  target tasks: ("repository")
+	// Full sync resets cursor and status and totalNumberOfRepos
+	// Partial sync only resets status (continues from existing cursor)
+	if (subscriptionTasks) {
+		if (syncType === "full") {
+			updateSubscriptionTasks.totalNumberOfRepos = null;
+			updateSubscriptionTasks.repositoryCursor = null;
+		}
+		updateSubscriptionTasks.repositoryStatus = null;
+	}
+
+	await subscription.update({
```

Comment:
This should only be called in the if statement above.  You don't need to call this when there's nothing to update.
You also don't need to use the spread operator when it's the only key in the object, just use `updateSubscriptionTasks` as the update object.